### ISSUE TITLE
App-route does not handle dots in file or pathnames

### ISSUE THREAD

**Issue by ghost** [suttacentral/suttacentral#3]
This is a big problem because many of our files have dots in them. I cannot load the AN or SN files the way things are now. Apparently it will work going through Firebase but that will be difficult for me with development. @blake-sc any ideas?

Filed bug report here:
https://github.com/Polymer/polyserve/issues/147

Stackoverflow question filed here:
https://stackoverflow.com/questions/45531951/polymer-2-0-iron-location-app-route-not-recognising-route

**Comment by blake-sc** [comment]
I'll look into modding app-route today, I don't think it'll be hard to fix.

**Comment by blake-sc** [comment]
I am unable to reproduce this issue. URLs with dots in them are working fine for me. It seems to be thought that this is an issue with polymer serve, maybe the version of polymer serve makes a difference? Mine is 1.3.1. Or can the issue be narrowed down? Are the routes not working at all for you or only in particular contexts?

Btw I tested by renaming "dn" to "d.n" and "dn1" to "dn1.1" everywhere, and checking if the site still worked when using those modified uids, and it did work.

**Comment by SamaneriVimala** [comment]
Did you try the MWE I posted on stackoverflow or did you try the next-sc full version? I've got 1.3.1 also.
Routes with any dots in are not working at all. Also tried dn1.1, etc. as you did.
And yes, as per the bug-report on Polyserve I posted above, it is indeed something to do with polyserve.

**Comment by blake-sc** [comment]
Well it's important to narrow the issue down. I'm having trouble working out if there is a problem with app-route at all, I don't think so, I pored over the source code and couldn't see any reference to '.', so the issue would seem to be strictly to do with the backend - polymer-cli in this case. I also tried modifying the tests to add urls with periods, these tests ran fine when executed in the browser, but crapped out when executed from polymer test, which strengthens the hypothesis that app-route is working fine.
